src/Diagnostics.RuntimeHost/Utilities/AzureSupportCenterInsightUtilities.cs

```diff
@@ -22,15 +22,11 @@ internal static class AzureSupportCenterInsightUtilites
             Value = "Go to applens to see more information about this insight."
         };
 
-        internal static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, Definition detector)
+        internal static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, ArmResourceType armResourceType,  Definition detector)
             where TResource : IResource
         {
-
-            // Logic in line before would have to be updated if we added more resource types. 
-            // TODO: May make sense to have resource type enums contain the resource type string
-            var resourceTypeString = context.Resource.ResourceType == ResourceType.App ? "sites" : "hostingEnvironments";
-            var issueCategory = context.Resource.ResourceType == ResourceType.App ? "Web App" : "App Service Environment";
-            var applensPath = $"subscriptions/{context.Resource.SubscriptionId}/resourceGroups/{context.Resource.ResourceGroup}/providers/Microsoft.Web/{resourceTypeString}/{context.Resource.Name}/detectors/{detector.Id}?startTime={context.StartTime}&endTime={context.EndTime}";
+            var applensPath = $"subscriptions/{context.Resource.SubscriptionId}/resourceGroups/{context.Resource.ResourceGroup}/providers/{armResourceType.Provider}/{armResourceType.ResourceTypeName}/{context.Resource.Name}/detectors/{detector.Id}?startTime={context.StartTime}&endTime={context.EndTime}";
+            var category = detector.Name.Length > 32 ? armResourceType.ResourceTypeName : detector.Name;
```

Comment:
Is there a limitation on the category length at ADS side?

Reply:
You bet there is! Before I just used "Web App" for every insight because I was worrying about the length but the guy complained so I figured using the detector name when possible would be better. Also just makes it clear what the insight is for.